### DISCUSSION TITLE
Confluence + Convertionnal Retrieval QA chain

### DISCUSSION THREAD

**Discussion by lde-net2** [FlowiseAI/Flowise#1993] [Q&A]
Hello,
<beginner here> don't shoot :)

i succeeded in connecting Confluence to a pinecone db and azure open ai, and it works like a charm, expecially after changing the text splitter from recursive chars to "markdown".
Now, i'd like to mention the source confluence pages in the answer,  they are stored in the database embedding, and i can see them in the flow in langsmith.

I added some instructions in the chain parameters,  from "With every answer, return the citations under a title "references"."  to more detailed, mentionning to use the metadata field "url"...  but no success
the bot returns sources links, but it is rewriting the urls to confluence ids that do not exist.
any idea / help ?

thanks a lot !!

Luc

**Comment by lde-net2** [comment]
Hello,

I am not sure what you mean.
Does it relate with my original question, which is how to return some metadata in the bot responses. (urls) ?

**Comment by mrkhachaturov** [comment]
Hello @lde-net2 . 
I am sorry , I was answering to another discussion

**Comment by mrkhachaturov** [comment]
@lde-net2 
Did you manage to add a link to the resource in the reply messages?

**Comment by lde-net2** [comment]
yes and no.  there is an issue where flowise stores the metadata page url as "url" instead of source
see here https://discord.com/channels/1087698854775881778/1216210489823203338
filled as bug as well https://discord.com/channels/1087698854775881778/1096265567058534461

but nobody answers there neither...

**Comment by lde-net2** [comment]
filled as bug in github https://github.com/FlowiseAI/Flowise/issues/2052

**Comment by lde-net2** [comment]
the confluence doc loader stores confluences pages url as metadata.url (langchain) while flowise  / ui  expect them as metadata.source
i hacked the ui chatmessage to take the metadata url ..
not clean, but it is HELL to get any doc/info/answer...
